src/renderer/component/userEmailVerify/view.jsx

```diff
@@ -1,104 +1,76 @@
 // @flow
 import * as React from 'react';
 import Button from 'component/button';
-import { Form, FormField, FormRow, Submit } from 'component/common/form';
 
 type Props = {
   cancelButton: React.Node,
-  errorMessage: ?string,
   email: string,
-  isPending: boolean,
-  onModal?: boolean,
-  verifyUserEmail: (string, string) => void,
-  verifyUserEmailFailure: string => void,
   resendVerificationEmail: string => void,
+  checkEmailVerified: () => void,
+  user: {
+    has_verified_email: boolean,
+  },
 };
 
-type State = {
-  code: string,
-};
-
-class UserEmailVerify extends React.PureComponent<Props, State> {
+class UserEmailVerify extends React.PureComponent<Props> {
   constructor(props: Props) {
     super(props);
-
-    this.state = {
-      code: '',
-    };
-
-    (this: any).handleSubmit = this.handleSubmit.bind(this);
+    this.emailVerifyCheckInterval = null;
     (this: any).handleResendVerificationEmail = this.handleResendVerificationEmail.bind(this);
   }
 
-  handleCodeChanged(event: SyntheticInputEvent<*>) {
-    this.setState({
-      code: String(event.target.value).trim(),
-    });
+  componentDidMount() {
+    this.emailVerifyCheckInterval = setInterval(() => this.checkIfVerified(), 5000);
```

Comment:
Five seconds?  Seems a bit long, we should look into doing this with websockets instead of polling.

Reply:
Yeah I figured that would be less hammering on the server. I agree.